### PR TITLE
build-sys: make autogen.sh stop when it cannot find one of the requir…

### LINKED ISSUE #3845: FreeBSD autoreconf error: Can't exec "aclocal": No such file or directory autoconf2.71/Autom4te/FileUtils.pm line 274
Trying to compile on FreeBSD 13.2, autoconf install, but I'm getting this error message:

```
Written by David J. MacKenzie and Akim Demaille.
##################################################################
automake: not found
##################################################################
#                        Generating files                        #
##################################################################
autoreconf2.71: export WARNINGS=
autoreconf2.71: Entering directory '.'
autoreconf2.71: configure.ac: not using Gettext
autoreconf2.71: running: aclocal --force -I m4
Can't exec "aclocal": No such file or directory at /usr/local/share/autoconf2.71/Autom4te/FileUtils.pm line 274.
autoreconf2.71: error: aclocal failed with exit status: 2
```

I've also installed `gettext` and restarted the terminal to see if ./autoreconf.sh would pick it up, since `autoreconf2.71: configure.ac: not using Gettext` precedes the actual error, but that didn't help. There's package of "Exuberant Ctags", so this isn't critical, as I can use that instead of "universal ctags". Just thought I'd report the issue. Thank you.

### PR DESCRIPTION
**Pull request by masatake**
…ed tools

Close #3845.

### CODE REVIEW

**Comment by leleliu008** [comment]
@masatake You can't do this. these commands do not need to be on $PATH, they can be used via setting environment variable.

**Comment by leleliu008** [comment]
```perl
my $autoconf    = $ENV{'AUTOCONF'}    || '/home/leleliu008/.ppkg/native/d7d2b14cbe2888e513a66893809b8ab2ba7986d1a970ee7bdcd2f797f4c42cf2/bin/autoconf';
my $autoheader  = $ENV{'AUTOHEADER'}  || '/home/leleliu008/.ppkg/native/d7d2b14cbe2888e513a66893809b8ab2ba7986d1a970ee7bdcd2f797f4c42cf2/bin/autoheader';
my $autom4te    = $ENV{'AUTOM4TE'}    || '/home/leleliu008/.ppkg/native/d7d2b14cbe2888e513a66893809b8ab2ba7986d1a970ee7bdcd2f797f4c42cf2/bin/autom4te';
my $automake    = $ENV{'AUTOMAKE'}    || 'automake';
my $aclocal     = $ENV{'ACLOCAL'}     || 'aclocal';
my $libtoolize  = $ENV{'LIBTOOLIZE'}  || 'libtoolize';
my $intltoolize = $ENV{'INTLTOOLIZE'} || 'intltoolize';
my $gtkdocize   = $ENV{'GTKDOCIZE'}   || 'gtkdocize';
my $autopoint   = $ENV{'AUTOPOINT'}   || 'autopoint';
my $make        = $ENV{'MAKE'}        || 'make';
```

If these environment variables aren't set, command would be search on PATH.

**Comment by masatake** [comment]
@leleliu008 I updated the error message.

File: autogen.sh [hunk 1]
```diff
@@ -8,6 +8,11 @@ for t in autoreconf aclocal pkg-config autoconf automake; do
 	if type $t; then
 		echo '------------------------------------------------------------------'
 		$t --version
+	else
+		echo '!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!'
+		echo "Please, install $t or reivse "'$PATH' 1>&2
```
Comment (k-takata) [inline autogen.sh:13]:
```suggestion
		echo "Please, install $t or revise "'$PATH' 1>&2
```

Comment (leleliu008) [reply autogen.sh:13]: I don't think we should show any warnings here. these warnings make me more confused. The error messages from autoreconf command is good engough.

Comment (masatake) [reply autogen.sh:13]: See #3845.

**Comment by masatake** [comment]
I'll withdraw this. @k-takata, @leleliu008 Sorry for making noise.

**Pull request closed without merging**